### ISSUE TITLE
Is Titan Dead?

### ISSUE THREAD

**Issue by codeclinic** [gambitph/Titan-Framework#426]
It'd be great if people could get some clarity on whether Titan is dead or not. A lot of people rely on this plugin and it's lack of updates is causing end users to uninstall our plugins and themes because they see that Titan hasn't been updated for a year or more.

I'm more than happy to come on board as maintainer &/or take over the plugin if you no longer have time to ma
intain it.

**Comment by codeclinic** [comment]
I'm trying to resurrect this framework. Please follow my repo and let's get this thing active again.

https://github.com/codeclinic/Abased-Titan-Framework

Thanks
Phill

**Comment by bfintal** [comment]
Hello, yes unfortunately I don’t have the time anymore to maintain the plugin :(

**Comment by codeclinic** [comment]
@bfintal Would you consider making me the maintainer or giving me the relevant admin permissions, then I can try and take it on?

**Comment by TreyOverton** [comment]
If you don't have time, then do those of us who committed to using the
plugin a favor and let Phill take it over. It's irresponsible to promote a
thing, get a bunch of people using it, then abandon it while also retaining
sole control.

Phill's already created a fork, but that adds complexity when getting to
the exact same goal anyway - a supported plugin without the known bugs.

**Comment by codeclinic** [comment]
Sadly it would appear the current maintainer wants to sell the plugin. For the amount of work involved in getting this up to a decent standard it's not conceivable for me to purchase this from @gambitph




On Mon, Dec 9, 2019 at 7:20 PM +0100, "Aaron Overton" <notifications@github.com> wrote:










If you don't have time, then do those of us who committed to using the


plugin a favor and let Phill take it over. It's irresponsible to promote a


thing, get a bunch of people using it, then abandon it while also retaining


sole control.





Phill's already created a fork, but that adds complexity when getting to


the exact same goal anyway - a supported plugin without the known bugs.





Blue skies,


-Aaron

**Comment by TreyOverton** [comment]
How much does he want for his open source plugin that anyone can just take
as is and market?

When WooThemes asked to buy Jigoshop, they offered an amount. Jigoshop got
insulted and demanded something more, like maybe 10x as much. So WooThemes
simply forked the code, paid them nothing, and hired their top programmer
away. Eventually, Automattic bought them out and WooCommerce became
effectively part of core.

Seems like a losing proposition and if that's the plan, then it's time to
fork and move on.

**Comment by codeclinic** [comment]
Don’t know how much he wants. I’m not prepared to ask. I’m a bit insulted to be honest. This thing has been dead for ages, and loads of people are stuck with it having spent time and money incorporating it. I spent ages trying to get some response about the plugin with no luck, then all of a sudden I get a message asking if I want to buy it.

 



How much does he want for his open source plugin that anyone can just take
as is and market?

When WooThemes asked to buy Jigoshop, they offered an amount. Jigoshop got
insulted and demanded something more, like maybe 10x as much. So WooThemes
simply forked the code, paid them nothing, and hired their top programmer
away. Eventually, Automattic bought them out and WooCommerce became
effectively part of core.

Seems like a losing proposition and if that's the plan, then it's time to
fork and move on.

Blue skies,
-Aaron

**Comment by codeclinic** [comment]
To be honest, the website wouldn’t be a problem for me to setup either and likewise it’d give a bit of traffic & seo value back to myself. I also have hosting infrastructure already setup. However, If you want to pay for Titan then I guess that’d be your privilege. Eitherway, I guess that's something for discussion further down the line if this goes anywhere.

**Comment by bfintal** [comment]
I'm sorry but please do not misconstrue my words. You do not need to be insulted since I did not ask you to buy it. Please read again what I sent you. Here it is so that everyone can see:

> I was actually planning on putting up Titan for sale since I couldn't maintain it anymore.

I did not post this reply in this issue and sent it in private since I didn't find it appropriate to post the plan here. But we're talking about the topic now anyway.

As for any details about this, to be honest this was just the plan but no specifics yet really. But probably when I get some free time again, I'd like to address the immediate problems of Titan and update it to a state that's stable again - no new features, just make it up-to-date. Then proceed from there.

I have poured tons of hours in this OSS / plugin, and never saw a dime from this project. It's been quite a while since we've used/updated it, so it's harder to maintain now. Realistically, I can't maintain this project since I have to focus on those that bring food to the table. But this was one of my first big project that gained somewhat of a traction, so Titan has always been special to me.

**Comment by TreyOverton** [comment]
Ben: totally valid perspective. But you're forgetting that people started depending on your work and now you can't maintain it, but also won't let others maintain it. You undo a great thing by a desire to not let go.

"When I get some free time again..." Last time that happened was October 2018. You just need to recognize that could totally be "never." And because it's all GPL and Open Source, you can either open it up and allow people to help you, or just face that at any moment someone will simply fork the code and run with it.

The option to be fully in control requires you to be actively maintaining. You can't, so you can either retain some control with help or have zero control because you just won't and people will take matters into their own hands without you. I think your watershed moment has arrived.

**Comment by pagelab** [comment]
It's really troublesome when things like these happens. I personally considered using Titan some years ago, but decided to choose another framework, that is still being actively maintained. I'm glad I have chosen the other option TBH.

But let's not forget that this is an Open Source project (GPLv2). So, by definition, you or anyone can fork it right now and build on top of it. You don't need special authorization from @bfintal to do this. It's the nature of FLOSS. Just remember to keep any copyright messages and refrain to use the “Titan” branding at all on the fork. The project's code and the brand are two separate things.

Also, anyone that uses a GPLv2 project like Titan should be aware that there's absolutely no guarantees in using it. This is explicitly said in the license's text.

So, the reality is that @bfintal is not obliged (legally, morally or otherwise) to give any kind of support or even maintain the project. It's his call.

**Comment by codeclinic** [comment]
@pagelab That's true and if you'd read the threads it would be clear to you that this has already been discussed. However, you would also notice that the issue is that @bfintal isn't prepared to let anyone else take over this project. Regardless of the listing on the WordPress plugins directory, it introduces a lot of issues with taking this forward if we are required to fork the project and then somehow encourage the community to follow us. This is especially true when considering the time, effort and to a degree trust that people have put into adopting this framework and using it in their products.

Simply handing the project over to new maintainers would enable the project to best honour @bfintal 's project& all his time and  hard work by allowing us to keep it alive. in its current form and thus minimise the impact on those invested in Titan. Thus, maximizsing the potential for titan to live on in some working form.

**Comment by bfintal** [comment]
“Isn’t prepared” is probably the perfect way to way to say it @codeclinic. I’d like to take a stab at it though to bring the project forward.

**Comment by milojennings** [comment]
I'd really like to see this project taken over and maintained by the community. It's a great framework. 
Unfortunately, my company is at a crossroads and will probably have to go a different direction with future projects due to the state of this project.

**Comment by milojennings** [comment]
> I personally considered using Titan some years ago, but decided to choose another framework, that is still being actively maintained. I'm glad I have chosen the other option TBH.

@pagelab Which framework did you go with?

**Comment by pagelab** [comment]
@milojennings I've been using [Kirki](https://wordpress.org/plugins/kirki/) for years now.

But with the [Full Site Editing](https://make.wordpress.org/design/handbook/focuses/full-site-editing/) and [Global Styles](https://make.wordpress.org/design/handbook/focuses/global-styles/) features being actively developed in WordPress Core right now, frameworks like this will be less and less needed.

**Comment by goodwinpress** [comment]
TF website is down
https://prnt.sc/rpghem

**Comment by TreyOverton** [comment]
#428 was reported and ignored, the fix is a few minutes. In this form, dead.

**Comment by codeclinic** [comment]
@AaronOverton @goodwinpress @pagelab @milojennings Myself and @davidsneighbour are working on a fork of Titan that will get Titan back as an active project and then add new features and optimisations. https://github.com/davidsneighbour/wp-titan-framework

**Comment by codeclinic** [comment]
@bfintal Benjamin, I'm working with a couple of people on getting Titan fixed up and in working order with a view to keeping the framework alive for as long as possible. Hopefully eve give it some new life & features & optimisations.

We are doing this because we invested time, effort and even money into adopting Titan Framework. We appreciate all the time work and money that you put into it, but since you no longer maintain the plugin it has left a lot of devs with a big problem.

If we were in a position to pay you then I know that we would. But, the truth is that this is an open source project and we just want to help out everyone including ourselves that are reliant on Titan for our plugins and themes.

Since you no longer maintaining or doing bug fixes, please please consider handing the plugin over to myself and the other people I am working with on resurrecting Titan.

We would be extremely grateful if you would consider passing over to us the WordPress plugin repo and allowing us to adopt it formerly. This would make it easy for existing plugins and themes to grab the latest version without needing to change code or ultimately turn to and embedded version.

Likewise or even alternatively if you could give us a copy of the documentation site, that would also be an amazing gesture. We'Re currently working on manually recreating the websites content/documentation using copies found via the Internet Archive Time Machine. IF we had the originals this would make things so much easier and quicker for everyone involved.

Thanks for your time and all the best with your Stackable project. The demo site looks great and I'm sure will attract a lot of interest for it.

P.s. Our goal is simply to get this working again for people reliant on the great work you did with Titan.

P.p.s For anyone interested in the new project come join us at: https://github.com/davidsneighbour/colossus-wp-options-framework

Kind Regards
Phill
